integration-tests/smoke/ccip_messaging_test.go

```diff
@@ -165,18 +172,138 @@ func Test_CCIPMessaging(t *testing.T) {
+func manuallyExecute(
+	ctx context.Context,
+	t *testing.T,
+	state ccdeploy.CCIPOnChainState,
+	destChain uint64,
+	out messagingTestCaseOutput,
+	sourceChain uint64,
+	e ccdeploy.DeployedEnv,
+	sender []byte,
+) {
+	merkleRoot := getMerkleRoot(
+		t,
+		state.Chains[destChain].OffRamp,
+		out.msgSentEvent.SequenceNumber,
+	)
+	messageHash := getMessageHash(
+		t,
+		state.Chains[destChain].OffRamp,
+		sourceChain,
+		out.msgSentEvent.SequenceNumber,
+		out.msgSentEvent.Message.Header.MessageId,
+	)
+	tree, err := merklemulti.NewTree(hashutil.NewKeccak(), [][32]byte{messageHash})
+	require.NoError(t, err)
+	proof, err := tree.Prove([]int{0})
+	require.NoError(t, err)
+	require.Equal(t, merkleRoot, tree.Root())
+
+	tx, err := state.Chains[destChain].OffRamp.ManuallyExecute(
+		e.Env.Chains[destChain].DeployerKey,
+		[]offramp.InternalExecutionReport{
+			{
+				SourceChainSelector: sourceChain,
+				Messages: []offramp.InternalAny2EVMRampMessage{
+					{
+						Header: offramp.InternalRampMessageHeader{
+							MessageId:           out.msgSentEvent.Message.Header.MessageId,
+							SourceChainSelector: sourceChain,
+							DestChainSelector:   destChain,
+							SequenceNumber:      out.msgSentEvent.SequenceNumber,
+							Nonce:               out.msgSentEvent.Message.Header.Nonce,
+						},
+						Sender:       sender,
+						Data:         []byte("hello CCIPReceiver with low exec gas"),
+						Receiver:     state.Chains[destChain].Receiver.Address(),
+						GasLimit:     big.NewInt(1),
+						TokenAmounts: []offramp.InternalAny2EVMTokenTransfer{},
+					},
+				},
+				OffchainTokenData: [][][]byte{
+					{},
+				},
+				Proofs:        proof.Hashes,
+				ProofFlagBits: boolsToBitFlags(proof.SourceFlags),
+			},
+		},
+		[][]offramp.OffRampGasLimitOverride{
+			{
+				{
+					ReceiverExecutionGasLimit: big.NewInt(200_000),
+					TokenGasOverrides:         nil,
+				},
+			},
+		},
+	)
+	_, err = deployment.ConfirmIfNoError(e.Env.Chains[destChain], tx, err)
+	require.NoError(t, err, "failed to send/confirm manuallyExecute tx")
+
+	newExecutionState, err := state.Chains[destChain].OffRamp.GetExecutionState(&bind.CallOpts{Context: ctx}, sourceChain, out.msgSentEvent.SequenceNumber)
+	require.NoError(t, err)
+	require.Equal(t, uint8(ccdeploy.EXECUTION_STATE_SUCCESS), newExecutionState)
+}
+
+func getMerkleRoot(
+	t *testing.T,
+	offRamp *offramp.OffRamp,
+	seqNr uint64,
+) (merkleRoot [32]byte) {
+	iter, err := offRamp.FilterCommitReportAccepted(nil)
+	require.NoError(t, err)
+	for iter.Next() {
+		for _, mr := range iter.Event.MerkleRoots {
+			if mr.MinSeqNr >= seqNr || mr.MaxSeqNr <= seqNr {
+				return mr.MerkleRoot
+			}
+		}
+	}
+	require.Fail(
+		t,
+		fmt.Sprintf("no merkle root found for seq nr %d", seqNr),
+	)
+	return merkleRoot
+}
+
+func getMessageHash(
+	t *testing.T,
+	offRamp *offramp.OffRamp,
+	sourceChainSelector,
+	seqNr uint64,
+	msgID [32]byte,
+) (messageHash [32]byte) {
+	iter, err := offRamp.FilterExecutionStateChanged(nil,
```

Comment:
`nil`? Shouldn't we pass the test context here?

Reply:
Yeah this is super ugly, lemme see what I can do. Also the block ranges are not specified meaning it'll search from genesis. Its OK for these simchains but it won't work on the real testnets